utilities/zookeeper/R/correctors.county.R

```diff
@@ -104,56 +108,62 @@ zyzzyva_county_corrections <- function(signals_list, ...){
+        (.data$value < -params$size_cut & 
              !is.na(.data$ststat) & !is.na(.data$ftstat)), # big negative
-        flag = .data$flag | # these allow smaller values to also be outliers if they are consecutive
-          (dplyr::lead(.data$flag) & !is.na(.data$ststat) & .data$ststat > params$sig_consec) | 
-          (dplyr::lag(.data$flag) & !is.na(.data$ststat) & .data$ststat > params$sig_consec) |
-          (dplyr::lead(.data$flag) & is.na(.data$ststat) & .data$ftstat > params$sig_consec) |
-          (dplyr::lag(.data$flag) & is.na(.data$ststat) & .data$ftstat > params$sig_consec),
-        flag = .data$flag & 
-          (.data$time_value < ymd(params$time_value_flag_date) | 
-             .data$value < -params$size_cut),
-        flag = .data$flag | 
-          (.data$time_value == "2020-11-20" & as.numeric(.data$geo_value) %/% 1000 == 22),
+      flag = .data$flag | # these allow smaller values to also be outliers if they are consecutive
+        (dplyr::lead(.data$flag) & !is.na(.data$ststat) & .data$ststat > params$sig_consec) | 
+        (dplyr::lag(.data$flag) & !is.na(.data$ststat) & .data$ststat > params$sig_consec) |
+        (dplyr::lead(.data$flag) & is.na(.data$ststat) & .data$ftstat > params$sig_consec) |
+        (dplyr::lag(.data$flag) & is.na(.data$ststat) & .data$ftstat > params$sig_consec),
+      flag = .data$flag & 
+        (.data$time_value < ymd(params$time_value_flag_date) | 
+            .data$value < -params$size_cut),
+      flag = .data$flag | 
+        (.data$time_value == "2020-11-20" & as.numeric(.data$geo_value) %/% 1000 == 22),
         #Louisiana backlog drop https://ldh.la.gov/index.cfm/newsroom/detail/5891
-        flag_bad_RI = ((substr(.data$geo_value,start = 1, stop = 2) == "44") &
+      flag_bad_RI = ((substr(.data$geo_value,start = 1, stop = 2) == "44") &
                          .data$value > 10 &
-                         abs(lag(.data$value) < params$integer_tol)),
-        excess = .data$value - na_replace(.data$smedian, .data$fmedian),
-        excess = floor(.data$excess - params$excess_cut*sign(.data$excess)*na_replace(.data$smad,.data$fmad)),
-        corrected = corrections_multinom_roll( # fix RI reporting problem
-          .data$value, .data$value, .data$flag_bad_RI, .data$time_value, 7),
-        corrected = corrections_multinom_roll( # for everywhere else
-          .data$corrected, .data$excess, (.data$flag & !.data$flag_bad_RI ),
-          .data$time_value, params$backfill_lag,
-          reweight=function(x) exp_w(x, params$backfill_lag))
+                         abs(lag(.data$value) < params$integer_tol))
       )
+        
+  if (unique(x$signal) == "confirmed_incidence_num") {
+    x <- x %>% mutate(corrected = corrections_multinom_roll( # fix RI reporting problem
+                        .data$value, .data$value, .data$flag_bad_RI, .data$time_value, 7),
+                      corrected = corrections_multinom_roll( # for everywhere else
```

Comment:
You shouldn't test with `unique()`. It can return a vector, and then the test will crash out with an error. Better to use `all()`.